odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+                                      self.src_radius, self.det_radius,
+                                      agrid=self.motion_grid,
+                                      dgrid=self.det_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    def __str__(self):
+        """`g.__str__() <==> str(g)`."""
+        return self.__repr__()  # TODO: prettify
+
+
+class ConeFlatGeometry(ConeBeamGeometry):
+    """Cone beam geometry in 3d with flat detector.
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``
+    and aligned tangential to the circle.
+
+    The motion parameter is the (1d) rotation angle parametrizing source and
+    detector positions.
+    """
+
+    def __init__(self, angle_intvl, dparams, src_rad, det_rad, agrid=None,
+                 dgrid=None, angle_offset=0, axis=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        angle_intvl : `Interval` or 1-dim. `IntervalProd`
+            The motion parameters
+        dparams : `Rectangle` or 2-dim. `IntervalProd`
+            The detector parameters
+        src_rad : `float`
+            Radius of the source circle, must be positive
+        det_rad : `float`
+            Radius of the detector circle, must be positive
+        agrid : 1-dim. `TensorGrid`, optional
+            A sampling grid for `angle_intvl`
+        dgrid : 2-dim. `TensorGrid`, optional
+            A sampling grid for `dparams`
+        angle_offset : `float`
+            Offset to the rotation angle in the azimuthal plane. Does not
+            affect the offset in z-direction.
+        axis : `int` or 3-element array
+            Defines the rotation axis via a 3-element vector or a single
+            integer referring to a standard axis
+        """
+        super().__init__(angle_intvl, src_rad, det_rad, agrid, angle_offset,
+                         axis)
+
+        if not (isinstance(dparams, IntervalProd) and dparams.ndim == 2):
+            raise TypeError('detector parameters {!r} are not an interval.'
+                            ''.format(dparams))
+
+        if dgrid is not None:
+            if not isinstance(dgrid, TensorGrid):
+                raise TypeError('detector grid {!r} is not a `TensorGrid` '
+                                'instance.'.format(dgrid))
+            if not dparams.contains_set(dgrid):
+                raise ValueError('detector grid {} not contained in detector '
+                                 'parameter interval {}.'
+                                 ''.format(dgrid, dparams))
+
+        self._detector = Flat2dDetector(dparams, dgrid)
+
+    @property
+    def detector(self):
+        """Detector of this geometry."""
+        return self._detector
+
+    def det_to_src(self, angle, dpar, normalized=True):
+        """Direction from a detector location to the source.
```

Comment:
This implementation is in the wrong place. The `ConeFlatGeometry` is intended as a parent class for any cone beam geometry with a flat detector, so at this point it is yet unknown how to get from the detector to the source. Only in the specialized guys with concrete acquisition curves this can be implemented. Now for the circular and helical cases, this happens to be the same, but still we cannot put it here since we may come up with other cases (e.g. flying focal spot?).